consensus/core/src/metrics.rs

```diff
@@ -77,24 +84,354 @@ const SIZE_BUCKETS: &[f64] = &[
     10_000_000.0,
 ]; // size in bytes
 
+const ERRORS_FROM_UNSIGNED_BLOCK_VERIFICATION: [&str; 6] = [
```

Comment:
I'd rather avoid using hardcoded strings to later compare them against error names. Maybe you can replace those with helper functions like:
```
fn is_from_unsigned_block_verification(err: &BlockError) -> bool {
    matches!(
        err,
        BlockError::WrongEpoch
            | BlockError::UnexpectedGenesisBlock
            | BlockError::InvalidAuthorityIndex
            | BlockError::SerializationFailure
            | BlockError::MalformedSignature
            | BlockError::SignatureVerificationFailure
    )
}
```